mungegithub/mungers/approval-handler.go

```diff
@@ -103,42 +104,32 @@ func (h *ApprovalHandler) Munge(obj *github.MungeObject) {
 
 		if len(fields) == 1 && strings.ToLower(fields[0]) == "/approve" {
 			approverSet.Insert(*c.User.Login)
-			continue
-		}
-
-		if len(fields) == 2 && strings.ToLower(fields[0]) == "/approve" && strings.ToLower(fields[1]) == "cancel" {
-			approverSet.Delete(*c.User.Login)
+		} else if len(fields) == 2 && strings.ToLower(fields[0]) == "/approve" && strings.ToLower(fields[1]) == "cancel" {
+			if approverSet.Has(*c.User.Login) {
+				approverSet.Delete(*c.User.Login)
+			}
 		}
 	}
 
+	//Checks that no valid approver has commented to cancel the approve label since last commit
+	//Checks that all files have an approver in the approverSet
 	for _, file := range files {
-		if !h.hasApproval(*file.Filename, approverSet, maxDepth) {
+		if !h.isApproved(file, approverSet) {
+			//no valid approvers
+			if obj.HasLabel(approvedLabel) {
+				glog.Infof("Canceling the approve label because %v file has no valid approvers", *file.Filename)
+				obj.RemoveLabel(approvedLabel)
+			}
 			return
 		}
 	}
+
+	//every file has been approved by a valid approver
 	obj.AddLabel(approvedLabel)
```

Comment:
Doesn't have to be in a function: if approved addLabel, else RemoveLabel.

Note that my patch needs to go in first if you don't want to check for the presence of the label before you add/remove.